implementing address

Sam09040 opened the pull request:
## added:
- address on schema;
- address db actions;
- address interface;
- address query;
- createAddress mutation;
- tests with createAddress mutation and address query;

## updated:
- separated user and address from db folder;
- put schema.prisma on db folder;
- put migrations folder on db folder;

Sam09040 committed:
implementing address

## added:
- address on schema;
- address db actions;
- address interface;
- address query;
- createAddress mutation;
- tests with createAddress mutation and address query;

## updated:
- separated user and address from db folder;
- put schema.prisma on db folder;
- put migrations folder on db folder;

racsandoval commented on src/data/graphql/query/address.query.ts:
```diff
@@ -0,0 +1,25 @@
+    });
+  }
+
+  const addresses = await getUserAddresses(userId);
+  return addresses;
```
```suggestion
  return getUserAddresses(userId);
```

racsandoval commented on test/address.test.ts:
```diff
@@ -0,0 +1,196 @@
+const port = process.env.PORT;
+const url = `http://localhost:${port}/`;
+
+describe.only('address testes', () => {
```
```suggestion
describe('address testes', () => {
```

racsandoval commented on test/address.test.ts:
```diff
@@ -0,0 +1,196 @@
+    await connectServer();
+    await connectDb();
+    await createUser(user);
+    const sam = await findUserByEmail('sam@example.com');
```
ao invés de criar e dar um find, a função `createUser` poderia retornar o usuário criado

SamuelSsc commented on test/address.test.ts:
```diff
@@ -0,0 +1,196 @@
+    });
+  });
+
+  it.only('should return success and show address', async () => {
```
```suggestion
  it('should return success and show address', async () => {
```

SamuelSsc commented on test/address.test.ts:
```diff
@@ -0,0 +1,196 @@
+    expect(res.state).to.equal(address[0].state);
+  });
+
+  it.only('should return 2 addresses', async () => {
```
```suggestion
  it('should return 2 addresses', async () => {
```

SamuelSsc commented on test/address.test.ts:
```diff
@@ -0,0 +1,196 @@
+  after('after', async () => {
+    await disconnectServer();
+    await deleteAllAddresses();
+    await deleteAllUsers();
```
Essas funções que limpam as tabelas deveriam estar num afterEach não?
Dessa forma que está vai ser limpo apenas uma vez após o describe todo ser rodado, e como um teste não pode depender do dado de outro teste ele deveria zerar o banco a cada it.

SamuelSsc replied:
Daria para simplificar, ao invés de chamar uma função para limpar cada tabela, ter uma função única que zere o banco

Sam09040 committed:
Fixes

## added:
- count function for user;
- optional skip and limit arguments on findAllUsers;
- beforeEach and afterEach on address test;

## removed:
- .only from describe and its on address test;
